uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java

```diff
@@ -820,6 +840,24 @@ public void discoverCoreDeviceBroker(String brokerId) {
         log.info("Added IoT Core broker as {} with endpoint {}:{}", brokerId, endpoint, IOT_CORE_MQTT_PORT);
     }
 
+
+    /**
+     * Set up IoT core broker port.
+     *
+     * @param brokerId broker name in tests
+     * @param port broker port in tests
+     */
+    @And("I set IoT Core broker as {string} with port {int}")
+    public void setBrokerPort(String brokerId, int port) {
```

Comment:
1. The name of method does not match what do name about unspecified broker but code is about only IoT Core broker.
2. That method probably is useless for your T20 because you try to connect to local broker. Maybe we should implement override port in existing address information for existing broker (does not matter local or IoT Core) just by brokerId

Reply:
fixed in 7d9ef728e6680002c72c9f03579e66403cd6a829

Reply:
No need to add to brokers,
Just take interate by address information for selected broker.
And change existing port number to specified.

Reply:
fixed in a39a3bb388a83630e9a3cf4c44a52ab0c2c8026e